benchmarks/src/main/java/org/elasticsearch/benchmark/compute/operator/BlockBenchmark.java

```diff
@@ -0,0 +1,174 @@
+                for (int blockIndex = 0; blockIndex < NUM_BLOCKS_PER_RUN; blockIndex++) {
+                    int[] values = new int[totalPositions()];
+                    int sum = 0;
+                    for (int i = 0; i < totalPositions(); i++) {
+                        values[i] = random.nextInt();
+                        sum += values[i];
+                    }
+                    checkSums[blockIndex] = sum;
+
+                    switch (blockKind) {
+                        case "array" -> {
+                            // TODO: bench with MVs and with nulls
+                            blocks[blockIndex] = new IntArrayBlock(
+                                values,
+                                totalPositions(),
+                                null,
+                                null,
+                                Block.MvOrdering.DEDUPLICATED_AND_SORTED_ASCENDING
+                            );
+                        }
+                        case "vector" -> {
+                            // TODO: more vector kinds
+                            IntVector vector = new IntArrayVector(values, totalPositions());
+                            blocks[blockIndex] = vector.asBlock();
+                        }
+                        default -> {
+                            throw new IllegalStateException();
+                        }
+                    }
+                }
+            }
+            case "long" -> {
+                for (int blockIndex = 0; blockIndex < NUM_BLOCKS_PER_RUN; blockIndex++) {
+                    long[] values = new long[totalPositions()];
+                    long sum = 0L;
+                    for (int i = 0; i < totalPositions(); i++) {
+                        values[i] = random.nextLong();
+                        sum += values[i];
+                    }
+                    checkSums[blockIndex] = sum;
+
+                    switch (blockKind) {
+                        case "array" -> {
+                            blocks[blockIndex] = new LongArrayBlock(
+                                values,
+                                totalPositions(),
+                                null,
+                                null,
+                                Block.MvOrdering.DEDUPLICATED_AND_SORTED_ASCENDING
+                            );
+                        }
+                        case "vector" -> {
+                            LongVector vector = new LongArrayVector(values, totalPositions());
+                            blocks[blockIndex] = vector.asBlock();
+                        }
+                        default -> {
+                            throw new IllegalStateException();
+                        }
+                    }
+                }
+            }
+            default -> {
+                throw new IllegalStateException();
+            }
+        }
+    }
+
+    @Benchmark
+    @OperationsPerInvocation(BlockBenchmark.NUM_BLOCKS_PER_RUN)
```

Comment:
Kind of a shame that the OperationsPerInvocation can't be some multiple of the block size. To really read the time things take you have to divide by the number of entries in the block.

I wonder if we're better off using a fixed number of entries in the block so we can get that? Not sure.

Reply:
Hm, maybe the block length turns out to not be important - then I'll just set it to a fixed number!

Reply:
Set it to a constant number and made it so we take the time per position instead of per block.